src/node_api.cc

```diff
@@ -824,11 +824,12 @@ void napi_module_register_cb(v8::Local<v8::Object> exports,
   // one is found.
   napi_env env = v8impl::GetEnv(context);
 
-  mod->nm_register_func(
-    env,
-    v8impl::JsValueFromV8LocalValue(exports),
-    v8impl::JsValueFromV8LocalValue(module),
-    mod->nm_priv);
```

Comment:
@kkoopa, @rvagg  I'm wondering if you have any insight as to why the module and exports were made available to modules in the past.  This change means that the addon won't have access to the module object and will only be able to replace the exports object as opposed to being able to add to it.

Reply:
No idea why. As far as I know, it is just the way it always has been (at least from Node v0.6 onwards).

Reply:
I think it was just for compatibility with JS-based CommonJS modules where you have both pre-created `exports` and `module` itself available.

Reply:
15508589a163b0c9f09ac608281f9ebb015d4deb

Reply:
Seems like it was done so native modules could also do `module.exports = function() {};`

Reply:
Ok from comments/backgroup it does not sound like we think the limitations will be a problem.  I guess worst case is we have to add a second register with additional parameters later on and best case we don't need them and stick with it as proposed here.  So once updates are added I'll take another look.

Reply:
One alternative that was discussed was to pre-emptively reserve a parameter in the register func signature in case the lack of module object availability during registration became a problem- that would allow us to deal with that situation in the future if it becomes a problem without causing a breaking change

Reply:
@digitalinfinity Even if it's needed in the future, it should be easy to add it as a separate function similar to `napi_get_global` (in which case `env` will be tied to that module). Although I don't see any good reason to expose such CommonJS implementation details to native modules for now.

Reply:
@RReverser That makes sense- for some reason I was imagining the fix would involve changing the signature of `napi_addon_register_func` but your approach makes sense- so I'm fine with this signature as is.